### PR TITLE
FreeBSD non-dev mount fstypes

### LINKED ISSUE #2781: Missing FreeBSD mount non-device fstypes
PR forthcoming.

### PR DESCRIPTION
**Pull request by steakknife**
Closes #2781

### CODE REVIEW

**Comment by steakknife** [comment]
ICLA signed.

**Comment by lamont-granquist** [comment]
is there any way to extend the `@new_resource.device != "none"` check to cover these use cases without having to maintain the long list of fstypes-without-devices?

**Comment by steakknife** [comment]
Not contain /

—

**Comment by steakknife** [comment]
Or \ 

—

**Comment by lamont-granquist** [comment]
Yeah, I can't imagine anyone would have a device without a path sep in it which referred to an actual location on disk.  I think if we skip this check we just skip some validation and just fail later as well, so even if there is some crazy edge condition out there, all we do is get a different error message?

**Comment by lamont-granquist** [comment]
Yeah, I'd take either patch, or both.  We might add the check for lack of path seps and then continue to maintain the list knowing that it can probably be dropped in the next major rev.

**Comment by steakknife** [comment]
Updated PR

—

File: lib/chef/provider/mount/mount.rb [hunk 1]
```diff
@@ -189,13 +189,14 @@ def network_device?
         end
 
         def device_should_exist?
-          ( @new_resource.device != "none" ) &&
-            ( not network_device? ) &&
-            ( not %w[ cgroup tmpfs fuse ].include? @new_resource.fstype )
+          @new_resource.device !~ FILE_SEP_REGEX &&
+           @new_resource.fstype !~ FILE_SEP_REGEX
```
Comment (lamont-granquist) [inline lib/chef/provider/mount/mount.rb]: does fstype take a path sometimes?

Comment (steakknife) [inline lib/chef/provider/mount/mount.rb]: Hit refresh. :)

**Comment by lamont-granquist** [comment]
needs specs now

**Comment by steakknife** [comment]
Mount provider has larger, non-idempotence issues interacting with `/sbin/mount` behavior.  Probably needs a rewrite because this isn't going to meet our requirements for this project, so closing.

```
devfs on /dev (devfs, local, multilabel)
bootpool on /bootpool (zfs, local, nfsv4acls)
zroot/tmp on /tmp (zfs, local, noatime, nosuid, nfsv4acls)
zroot/usr/home on /usr/home (zfs, local, noatime, nfsv4acls)
zroot/usr/ports on /usr/ports (zfs, local, noatime, nosuid, nfsv4acls)
zroot/usr/src on /usr/src (zfs, local, noatime, nfsv4acls)
zroot/var/crash on /var/crash (zfs, local, noatime, noexec, nosuid, nfsv4acls)
zroot/var/log on /var/log (zfs, local, noatime, noexec, nosuid, nfsv4acls)
zroot/var/mail on /var/mail (zfs, local, nfsv4acls)
zroot/var/tmp on /var/tmp (zfs, local, noatime, nosuid, nfsv4acls)
fdescfs on /dev/fd (fdescfs)
fdescfs on /dev/fd (fdescfs)
fdescfs on /dev/fd (fdescfs)
fdescfs on /dev/fd (fdescfs)
```

**Pull request closed without merging**